lib/ExpensiMark.js

```diff
@@ -330,16 +341,15 @@ export default class ExpensiMark {
      * @returns {String}
      */
     htmlToMarkdown(htmlString) {
-        let generatedMarkdown = _.unescape(htmlString);
-
+        let generatedMarkdown = htmlString;
```

Comment:
Won't this affect other conversions, as we would have regex for processing unescaped strings

Reply:
I didn't understand completely. Do you mean that other tags parsing in HTM=> MD conversion or All the conversions HTML=> MD, MD=> HTML?

Reply:
Yes other tags parsing in HTML->MD

Reply:
@MonilBhavsar I moved this down after the regex match based on two conditions.

1. First, If HTML contains an escaped string that can contain escaped tags, then that string is not parsed at HTML. it was escaped so that it shows as a raw string. Thus it does not have an effect on HTML parsing.
For example. If you can to show in the UI `<b>Rajat</b>` as a string in HTML. 
You can't write it as `<p> <b>Rajat</b> </p>`. It would be shows as <b>Rajat</b>. Instead you have to wirte `<p>&lt;b&gt;Rajat&lt;/b&gt;`.

2. In href, URLs, `"` could be used as &quot; to keep the HTML valid. But if unescape it before matching them it becomes `"` which breaks the URL parsing.
for example <a href="http://a-&quot;url">test</a>.
here url will be parsed as http://a- if we unescape ealier.

3. I added this _.unesacpe earlier to show the chars correctly. 
take the example from 1, `<p>&lt;b&gt;Rajat&lt;/b&gt;</p>`. will give as `&lt;b&gt;Rajat&lt;/b&gt;` after parsing but on screen you want to see `<b/>Rajat</b>`.

Hope it answers your question. Please let me know if I missed some cases. I will rethink that.

Reply:
Not sure If this is right, I follow that, 

1. If html contains escaped characters, it's a raw string and it won't be parsed here? Right now if I write `&lt;p&gt;&lt;b&gt;Monil&lt;/b&gt;&lt;/p&gt;` then I see `<p><b>Monil</b></p>`

2. Makes sense, that's why we are not unescaping it before. 

3. Sorry can't follow this.

My take, Let' say I type `<em>Monil</em>` then it should parse italics regex rule and show as `_Monil_`(_Monil_). And when I copy this, I get `&lt;em&gt;Monil&lt;/em&gt;`. So this is not working as expected right now, right?

Reply:
Sorry for the typos in the earlier message, 
1. If you paste `<em>Monil</em>`, you will get `_Monil_`.  But if you type this `<em>Monil</em>` its is taken as HTML , we don't support HTML markup in composer. It eould be saved as `&lt;em&gt;Monil&lt;/em&gt;`.

2. take the example from 1, if you pase HTML as `<p>&lt;b&gt;Rajat&lt;/b&gt;</p>`. Here I have used escaping to differentiate tages from text. But at the source location In UI `&lt;b&gt;Rajat&lt;/b&gt;` is shown as `<b>Rajat</b>` raw text.

Now if you parse `<p>&lt;b&gt;Rajat&lt;/b&gt;</p>`, it will give as `&lt;b&gt;Rajat&lt;/b&gt;` text in expensify. But as a user you don't want to see this in composer. A user would like to see it as  `<b/>Rajat</b>` But it does not mean that it is HTML. Only for this reason we have added unescaping.

@MonilBhavsar Where are copying from?

Reply:
Makes sense now. Thanks.
Messages with html tags will br parsed straight away by our regex rules.
Messages with escaped characters won't, we'll just unescape them and display it.

Right?
If I copy and paste - `<em>Monil</em>` and `<b>Monil</b>` from github and send message in App, they are sent as HTML raw code.
But If I copy that sent message and paste it again, I see converted forms like `_Monil_` and `_*Monil*_` in composer

Reply:
it does not matter what is shown in UI, the main thing is what is copied as HTML. Test what do you have `text/html` http://madebyevan.com/clipboard-test/ .

Reply:
Okay, works when type is `text/html`